commit 3fbf17b6271ca389cd4de1e33d6ec33708261228
Author: Andrew Cagney <cagney@sourceware.org>
Date:   Sat Jul 17 13:06:46 2010 -0400

    Add --enable-gnome; make gnome interface optional.
    
    2010-07-17  Andrew Cagney  <cagney@gnu.org>
    
    	* configure.ac: Add --enable-gnome option; only check for
    	java-gnome when enabled.
    
    	* Makefile.am (manpages manpages/index.html): Only include
    	frysk-gui when ENABLE_GNOME.

diff --git a/frysk-top/ChangeLog b/frysk-top/ChangeLog
index 00498f2..0c72180 100644
--- a/frysk-top/ChangeLog
+++ b/frysk-top/ChangeLog
@@ -1,3 +1,11 @@
+2010-07-17  Andrew Cagney  <cagney@gnu.org>
+
+	* configure.ac: Add --enable-gnome option; only check for
+	java-gnome when enabled.
+
+	* Makefile.am (manpages manpages/index.html): Only include
+	frysk-gui when ENABLE_GNOME.
+
 2008-06-06  Rick Moseley  <rmoseley@redhat.com>
 
 	* NEWS: Add bz #'s; add item on hover-over variable.
diff --git a/frysk-top/Makefile.am b/frysk-top/Makefile.am
index d852fba..49444fb 100644
--- a/frysk-top/Makefile.am
+++ b/frysk-top/Makefile.am
@@ -41,30 +41,35 @@
 include common/Makefile.rules
 
 
-SUBDIRS = \
-	frysk-imports \
-	frysk-sys \
-	frysk-core \
-	frysk-gtk \
-	frysk-gui
+SUBDIRS =
+SUBDIRS += frysk-imports
+SUBDIRS += frysk-sys
+SUBDIRS += frysk-core
+if ENABLE_GNOME
+SUBDIRS += frysk-gtk
+SUBDIRS += frysk-gui
+endif
 
 # Note the dir/. is needed, causes the file copy to put the contents
 # into the top level directory.
 
-SOURCEDIRS = \
-	$(patsubst %,$(srcdir)/%/frysk,$(SUBDIRS)) \
-	$(patsubst %,$(top_builddir)/%/frysk,$(SUBDIRS)) \
-	$(srcdir)/frysk-sys/lib \
-	$(top_builddir)/frysk-sys/lib \
-	$(srcdir)/frysk-sys/inua \
-	$(top_builddir)/frysk-sys/inua \
-	$(srcdir)/frysk-imports/getopt/src/getopt/gnu \
-	$(srcdir)/frysk-imports/jline/src/jline \
-	$(srcdir)/frysk-imports/junit/src/junit \
-	$(srcdir)/frysk-core/frysk/pkglibdir/. \
-	$(srcdir)/frysk-core/frysk/bindir/. \
-	$(srcdir)/frysk-gui/frysk/bindir/. \
-	$(empty)
+SOURCEDIRS =
+SOURCEDIRS += $(patsubst %,$(srcdir)/%/frysk,$(SUBDIRS))
+SOURCEDIRS += $(patsubst %,$(top_builddir)/%/frysk,$(SUBDIRS))
+SOURCEDIRS += $(srcdir)/frysk-sys/lib
+SOURCEDIRS += $(top_builddir)/frysk-sys/lib
+SOURCEDIRS += $(srcdir)/frysk-sys/inua
+SOURCEDIRS += $(top_builddir)/frysk-sys/inua
+SOURCEDIRS += $(srcdir)/frysk-imports/getopt/src/getopt/gnu
+SOURCEDIRS += $(srcdir)/frysk-imports/jline/src/jline
+SOURCEDIRS += $(srcdir)/frysk-imports/junit/src/junit
+SOURCEDIRS += $(srcdir)/frysk-core/frysk/pkglibdir/.
+SOURCEDIRS += $(srcdir)/frysk-core/frysk/bindir/.
+SOURCEDIRS += $(srcdir)/frysk-gui/frysk/bindir/.
+
+if ENABLE_GNOME
+SOURCEDIRS += $(srcdir)/frysk-gui/frysk/bindir/.
+endif
 
 
 # Generate html manpages.  Use same xml sources, but generate html.
@@ -78,6 +83,7 @@ TEMPLATE = $(srcdir)/htdocs/template.html
 manpages manpages/index.html:
 	rm -rf manpages
 	mkdir -p manpages
+if ENABLE_GNOME
 	XMLTO=$(XMLTO) \
 	sh $(srcdir)/frysk-common/manpages.sh \
 		$(TEMPLATE) \
@@ -89,6 +95,17 @@ manpages manpages/index.html:
 		frysk-gui/frysk/bindir/*.xml \
 		-"Test framework" \
 		*/frysk/pkglibdir/*.xml
+else
+	XMLTO=$(XMLTO) \
+	sh $(srcdir)/frysk-common/manpages.sh \
+		$(TEMPLATE) \
+		-"Overview" \
+		frysk-common/frysk.xml \
+		-"Command Line Utilities" \
+		frysk-core/frysk/bindir/*.xml \
+		-"Test framework" \
+		*/frysk/pkglibdir/*.xml
+endif
 	mv manpages/index.new manpages/index.html
 
 # Generate JAVADOC documentation.
diff --git a/frysk-top/configure.ac b/frysk-top/configure.ac
index 6e1d234..bf122ce 100644
--- a/frysk-top/configure.ac
+++ b/frysk-top/configure.ac
@@ -43,11 +43,24 @@ sinclude(common/version.ac)
 AC_INIT(frysk,defn([FRYSK_VERSION]))
 m4_include([common/frysk-common.ac])
 
+AC_ARG_ENABLE(gnome,
+AC_HELP_STRING(
+	[--enable-gnome],
+	[enable gnome user interface (default no)]),
+[enable_gnome=$enableval],
+[enable_gnome=no])
+AM_CONDITIONAL([ENABLE_GNOME], [test x$enable_gnome = xyes])
+
 AM_INIT_AUTOMAKE([subdir-objects foreign no-installinfo no-exeext no-dist])
 AC_CONFIG_FILES([Makefile])
-AC_CONFIG_SUBDIRS([frysk-imports frysk-sys frysk-core frysk-gtk frysk-gui])
+
+AC_CONFIG_SUBDIRS([frysk-imports frysk-sys frysk-core])
+if test x$enable_gnome = xyes; then
+  AC_CONFIG_SUBDIRS([frysk-gtk frysk-gui])
+fi
 
 # Need to call PKG_CHECK_MODULES to get PKG_CONFIG set; used by javadoc.
+if test x$enable_gnome = xyes ; then
 PKG_CHECK_MODULES(FRYSK_GNOME, [gtk+-2.0 >= 2.7.0 gtk2-java >= 2.7.0 
 glade-java >= 2.7.0 glib-java >= 0.2 cairo-java >= 1.0 vte-java >= 
 0.11.11 pango >= 1.10.0 pangox >= 1.10.0 pangoft2 >= 1.10.0])
@@ -55,5 +68,6 @@ FRYSK_PKG_MODULE_VARIABLE(FRYSK_GNOME_JARS, classpath, [gtk+-2.0 >= 2.7.0
 gtk2-java >= 2.7.0 glade-java >= 2.7.0 glib-java >= 0.2 
 cairo-java >= 1.0 vte-java 
 >= 0.11.11 pango >= 1.10.0 pangox >= 1.10.0 pangoft2 >= 1.10.0])
+fi
 
 AC_OUTPUT
